plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1973,7 +1969,7 @@ public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(C
                         (columnHandle.getType() == UUID && !(domain.isOnlyNull() || domain.getValues().isAll()))) {
                     unsupported.put(columnHandle, domain);
                 }
-                else if (canEnforceColumnConstraintInSpecs(typeOperators, icebergTable, partitionSpecIds, columnHandle, domain)) {
+                else if (canEnforceColumnConstraintInSpecs(typeManager.getTypeOperators(), icebergTable, partitionSpecIds, columnHandle, domain)) {
```

Comment:
Why not inject the `TypeOperators`  instance in both classes through Guice instead of depending on the getter of `typeManager` ?

Reply:
It would require additional bindings. Let me merge as-is.